lib/index.js

```diff
@@ -469,6 +471,40 @@ class AuthenticationPlugin {
     return this.getUsersRepository().delete(user._id, {refresh: 'wait_for'});
   }
 
+  /**
+   * Search users by ES query concerning exclusively their username
+   *
+   * @param {Object} query
+   *
+   * @returns {Promise.<{ scrollId, hits, aggregations, total }>}
+   */
+  async search (query) {
+    debug('exists(query = %j)', query);
+
+    const sanitizedQuery = objectRecursiveMap(query, (key) => {
+      if (key === 'username') {
+        return { key: '_id' };
+      }
+
+      const forbiddenProperties = Object.keys(storageMapping.users.properties);
+      
+      if ( forbiddenProperties.find(property => property === key ) ) {
```

Comment:
Nitpicking: coding standard
```suggestion
      if (forbiddenProperties.find(property => property === key)) {
```

Also, 
 - I think you can use `Array.includes` instead of `Array.find`
 - I would prefer a whitelist than a blacklist (you mad BLM?), because users may try to use wildcards for example and thus be able to match other properties (https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html)